### PR TITLE
Feature iops 2042 profile comparison nhs england profiles vs nhs digital profiles

### PR DESCRIPTION
**Pull request by KazeemHamzat**
Feature iops 2042 profile comparison nhs england profiles vs nhs digital profiles

### CODE REVIEW

File: structuredefinitions/England-Location.xml
Comment (digitalhealthmartin) [inline structuredefinitions/England-Location.xml:1]: According to spreadsheet, Location.type.extension:locationRole is missing max=1 but this isn't reflected here.

File: structuredefinitions/England-MedicationRequest.xml
Comment (digitalhealthmartin) [inline structuredefinitions/England-MedicationRequest.xml:1]: isModifier=false is there but in spreadsheet it states it should be mustSupport=true for the extensions

Comment (digitalhealthmartin) [reply structuredefinitions/England-MedicationRequest.xml:1]: On the spreadsheet it was raised about the min cardinality of MedicationRequest.subject.reference. It's currently min=1 in NHSE but it was not this in NHSD, why is this the case?

Comment (digitalhealthmartin) [reply structuredefinitions/England-MedicationRequest.xml:1]: On the spreadsheet it was raised about the min cardinality of MedicationRequest.dispenseRequest.performer.identifier. It's currently min=1 in NHSE but it was not this in NHSD, why is this the case?

File: structuredefinitions/England-ServiceRequest.xml [hunk 1]
```diff
@@ -205,10 +190,6 @@
     <element id="ServiceRequest.code.coding:SNOMEDProcedureCode">
       <path value="ServiceRequest.code.coding" />
       <sliceName value="SNOMEDProcedureCode" />
-      <binding>
-        <strength value="preferred" />
-        <valueSet value="https://fhir.hl7.org.uk/ValueSet/UKCore-ProcedureCode" />
```
Comment (digitalhealthmartin) [inline structuredefinitions/England-ServiceRequest.xml:210]: Based on your discussions with UK Core, this was was advised?

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]

File: structuredefinitions/England-Location.xml
Comment (digitalhealthmartin) [reply structuredefinitions/England-Location.xml:1]: Also, why is there only one slice in NHSE in Location.identifier but there are two in NHSD?

Comment (digitalhealthmartin) [reply structuredefinitions/England-Location.xml:1]: And also the cardinality max in NHSD it * but you're proposing 1 in NHSE

Comment (KazeemHamzat) [reply structuredefinitions/England-Location.xml:1]: 1. max=1 has been added
2.  min=1, line 52 will resolve slicing issue
3.

Comment (KazeemHamzat) [reply structuredefinitions/England-Location.xml:1]: max=1 changed to max=*

File: structuredefinitions/England-ServiceRequest.xml (hunk 1 above)
Comment (KazeemHamzat) [reply structuredefinitions/England-ServiceRequest.xml:210]: Binding was removed to revert to present state in NHSDigital-ServiceRequest. 
Note: ServiceRequest.code.coding:SNOMEDProcedureCode has no binding. IOPS-2086 has been raised(needs DA discussion for resolution)

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]
I looked at my comments for Location and although you marked them as resolved and commented that you updated them I don't see that reflected in the code I don't think. I'll look at it again and the other profiles once you can show me the fixes are actually pushed in.

File: structuredefinitions/England-Task.xml
Comment (SheenaSharma) [inline structuredefinitions/England-Task.xml:1]: Checked with the changes as mentioned in the ticket (NHSDigital-Task vs England-Task) and it looks fine to me.

**Comment by KazeemHamzat** [comment]
> I looked at my comments for Location and although you marked them as resolved and commented that you updated them I don't see that reflected in the code I don't think. I'll look at it again and the other profiles once you can show me the fixes are actually pushed in.

Response to earlier comment on England-Location are in code lines 43 and 75.
1. set max=' * ' in Forge deletes the code.
2. set max= ' 1 '  for Location.type.extension:locationRole is now appearing.

File: structuredefinitions/England-Task.xml
Comment (KazeemHamzat) [reply structuredefinitions/England-Task.xml:1]: Resolution of England-ServiceRequest Ticket is not in this PR. The reason the status is still in-progress. 
The changes on England-ServiceRequest in this PR are Forge recommendations and a reversal to present state of NHSDigital-ServiceRequest.
England-ServiceRequest Ticket, [IOPS-2086](https://nhsd-jira.digital.nhs.uk/browse/IOPS-2086) has been scheduled for DA discussion and resolution.

File: structuredefinitions/England-ServiceRequest.xml
Comment (SheenaSharma) [inline structuredefinitions/England-ServiceRequest.xml:1]: ServiceRequest profile issues will be resolved/assigned in different feature branch.

File: structuredefinitions/England-PractitionerRole.xml
Comment (SheenaSharma) [inline structuredefinitions/England-PractitionerRole.xml:1]: Looks fine to me, as it doesn't have any major changes except removing <min value="1" />.

File: structuredefinitions/England-Practitioner.xml
Comment (SheenaSharma) [inline structuredefinitions/England-Practitioner.xml:1]: Looks fine, as it doesn't have any major changes except removing <min value="0" />.

File: structuredefinitions/England-Organization.xml
Comment (SheenaSharma) [inline structuredefinitions/England-Organization.xml:1]: Looks fine. Changes are done which are mentioned in the ticket.

File: structuredefinitions/England-OperationOutcome.xml
Comment (SheenaSharma) [inline structuredefinitions/England-OperationOutcome.xml:1]: Looks fine, changes are done as mentioned in the ticket.

File: structuredefinitions/England-MessageHeader.xml
Comment (SheenaSharma) [inline structuredefinitions/England-MessageHeader.xml:1]: MessageHeader profile issues will be handled/resolved in different feature branch.

File: structuredefinitions/England-Patient.xml
Comment (SheenaSharma) [inline structuredefinitions/England-Patient.xml:1]: Forge recommendations, i.e. <min value="0" />are done.
Cardinality issues are to be treated by the UK Core Team.

File: structuredefinitions/England-Observation.xml
Comment (SheenaSharma) [inline structuredefinitions/England-Observation.xml:1]: Changes are done which are mentioned in the ticket.
Only issue with indentation.

File: structuredefinitions/England-MedicationRequest.xml
Comment (SheenaSharma) [inline structuredefinitions/England-MedicationRequest.xml:1]: MedicationRequest.medication[x]: Remove redundant code lines

This change needs to be done. Please remove the redundant code lines from MedicationRequest.medication[x].

**Review by digitalhealthmartin — APPROVED** [review]

File: structuredefinitions/England-MedicationRequest.xml
Comment (KazeemHamzat) [reply structuredefinitions/England-MedicationRequest.xml:1]: Redundant codes have been removed.

**Review by SheenaSharma — APPROVED** [review]

**Review by SheenaSharma — APPROVED** [review]

**Pull request merged**